Tech-Lab-On-Campus/Resources/Git-Commands.md

```diff
@@ -0,0 +1,137 @@
+- **Diffs** - Every time you commit there are differences between the project from the last save point to the new one. These changes are called differences, or “diffs” for short. You can compare changes between commits by looking at the diffs.
+- **Merge Conflict** - A conflict occurs when two parties change the same file and there's no automatic way to reconcile the changes. You must resolve the change by selecting which change wins.
+- **Clone** - When you clone a Git repository, you create a local copy to work on and your version will match the contents of the repository at the time you cloned it. For instance, you can clone a repository on GitHub to your computer where you can compile it from source, modify it, and more.
+- **Pull** - If you've been working on a clone project, you may need to catch up with changes others have made by pulling, which changes revisions from the repository to your local copy.
+- **Forking** - Another way to copy a repository is called “forking,” and is more like a fork in a tree. Unlike when you clone and make a copy on your computer, forking involves making a new repository, typically on GitHub or whatever hosting site you're using. A fork has all the contents of the original, but you are now the owner of this new repository and you're starting a new development project. You're copying a project when you fork and you're working on a project when you clone.
+- **Branch** - Branches let you create alternate versions of your project. When you branch you can work on new features and try new ideas without messing with your main work. Consider this – your main work may have many names and, like a tree trunk, it’s commonly called the “master,” “master branch,” or “baseline.” The trunk or master is where your team keeps its primary project development and if it’s messed with, your team will suffer. Thus, if you plan to work on something that will take a while and may break your build, be sure to branch so you don't interfere with the work of others on your team.
+- [**Open Source**](https://opensource.com/resources/what-open-source) - The term open source refers to something people can modify and share because its design is publicly accessible.
+
+There are other distributed version control systems that are widely used, for example Mercurial. Instead of typing `git` before commands, you would use `hg` for this version control system. The three most popular [DVCS (Distributed Version Control Systems)](https://www.atlassian.com/blog/software-teams/version-control-centralized-dvcs#:~:text=Distributed%20Version%20Control,-In%20the%20past&text=The%20three%20most%20popular%20of,on%20their%20own%20hard%20drive.) are Mercurial, Git, and Bazaar. Here, we will solely focus on Git.
```

Comment:
```suggestion
There are other distributed version control systems that are widely used, for example Mercurial. Instead of typing `git` before commands, you would use `hg` for this version control system. The three most popular DVCS (Distributed Version Control Systems) are Mercurial, Git, and Bazaar. Here, we will solely focus on Git.
```